### PR TITLE
GH1704: Move CakeConsole & CakeBuildLog to Cake.Core

### LINKED ISSUE #1704: Move CakeConsole & CakeBuildLog to Cake.Core and made CakeConfiguration public

### PR DESCRIPTION
**Pull request by devlead**
* Make CakeConfiguration public
* Fixes #1704

### CODE REVIEW

File: src/Cake.Core/Configuration/CakeConfiguration.cs [hunk 1]
```diff
@@ -7,15 +7,27 @@
...
+    /// Represents the Cake configuration.
+    /// </summary>
+    public sealed class CakeConfiguration : ICakeConfiguration
     {
         private readonly Dictionary<string, string> _lookup;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CakeConfiguration"/> class.
+        /// </summary>
+        /// <param name="lookup">The lookup.</param>
```
Comment (patriksvensson) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: Perhaps change to something `The initial configuration table.`?

File: src/Cake.Core/Configuration/CakeConfiguration.cs [hunk 2]
```diff
@@ -7,15 +7,27 @@
...
         public CakeConfiguration(IDictionary<string, string> lookup)
         {
             _lookup = new Dictionary<string, string>(lookup, StringComparer.OrdinalIgnoreCase);
         }
 
+        /// <summary>
+        /// Gets the value that corresponds to the specified key.
+        /// </summary>
+        /// <param name="key">The key.</param>
```
Comment (patriksvensson) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: `The configuration key`

File: src/Cake.Core/Configuration/CakeConfiguration.cs [hunk 3]
```diff
@@ -7,15 +7,27 @@
...
+        /// Initializes a new instance of the <see cref="CakeConfiguration"/> class.
+        /// </summary>
+        /// <param name="lookup">The lookup.</param>
         public CakeConfiguration(IDictionary<string, string> lookup)
         {
             _lookup = new Dictionary<string, string>(lookup, StringComparer.OrdinalIgnoreCase);
         }
 
+        /// <summary>
+        /// Gets the value that corresponds to the specified key.
```
Comment (patriksvensson) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: `Gets the value that corresponds to the specified configuration key.`

File: src/Cake.Core/Configuration/CakeConfiguration.cs [hunk 4]
```diff
@@ -7,15 +7,27 @@
...
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <returns>The value for the specified key, or <c>null</c> if key doesn't exists.</returns>
```
Comment (patriksvensson) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: `The value for the specified configuration key`

File: src/Cake.Core/Diagnostics/CakeBuildLog.cs [hunk 5]
```diff
@@ -4,20 +4,30 @@
...
-using Cake.Core.Diagnostics;
-using Cake.Diagnostics.Formatting;
+using Cake.Core.Diagnostics.Formatting;
 
-namespace Cake.Diagnostics
+namespace Cake.Core.Diagnostics
 {
-    internal sealed class CakeBuildLog : ICakeLog
+    /// <summary>
+    /// Represents Cake build log.
```
Comment (patriksvensson) [inline src/Cake.Core/Diagnostics/CakeBuildLog.cs]: `The default Cake build log.`

File: src/Cake.Core/Configuration/CakeConfiguration.cs [hunk 6]
```diff
@@ -7,15 +7,27 @@
 
 namespace Cake.Core.Configuration
 {
-    internal sealed class CakeConfiguration : ICakeConfiguration
+    /// <summary>
+    /// Represents the Cake configuration.
```
Comment (patriksvensson) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: Perhaps: `The default implementation of the Cake configuration.`

File: src/Cake.Core/CakeConsole.cs [hunk 7]
```diff
@@ -3,11 +3,13 @@
...
 
 using System;
-using Cake.Core;
 
-namespace Cake
+namespace Cake.Core
 {
-    internal sealed class CakeConsole : IConsole
+    /// <summary>
+    /// Represents a console.
```
Comment (patriksvensson) [inline src/Cake.Core/CakeConsole.cs]: Perhaps: `The default console implementation.`

File: src/Cake.Core/CakeReportPrinter.cs [hunk 8]
```diff
@@ -6,22 +6,33 @@
...
 using System.Linq;
-using Cake.Core;
 using Cake.Core.Diagnostics;
 
-namespace Cake
+namespace Cake.Core
 {
-    internal sealed class CakeReportPrinter : ICakeReportPrinter
+    /// <summary>
+    /// Represents a report printer.
```
Comment (patriksvensson) [inline src/Cake.Core/CakeReportPrinter.cs]: Perhaps `The default report printer.`

File: src/Cake.Core/Configuration/CakeConfiguration.cs (hunk 6 above)
Comment (devlead) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: sure

File: src/Cake.Core/Configuration/CakeConfiguration.cs (hunk 1 above)
Comment (devlead) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: sure

File: src/Cake.Core/Configuration/CakeConfiguration.cs (hunk 3 above)
Comment (devlead) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: sure

File: src/Cake.Core/Configuration/CakeConfiguration.cs (hunk 2 above)
Comment (devlead) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: sure

File: src/Cake.Core/Configuration/CakeConfiguration.cs (hunk 4 above)
Comment (devlead) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: Sure

File: src/Cake.Core/Diagnostics/CakeBuildLog.cs (hunk 5 above)
Comment (devlead) [inline src/Cake.Core/Diagnostics/CakeBuildLog.cs]: sure

File: src/Cake.Core/CakeConsole.cs (hunk 7 above)
Comment (devlead) [inline src/Cake.Core/CakeConsole.cs]: sure

File: src/Cake.Core/CakeReportPrinter.cs (hunk 8 above)
Comment (devlead) [inline src/Cake.Core/CakeReportPrinter.cs]: sure

**Comment by devlead** [comment]
@patriksvensson addressed feedback.

File: src/Cake.Core/Configuration/CakeConfiguration.cs [hunk 9]
```diff
@@ -7,15 +7,27 @@
 
 namespace Cake.Core.Configuration
 {
-    internal sealed class CakeConfiguration : ICakeConfiguration
+    /// <summary>
+    /// The initial configuration table.
```
Comment (patriksvensson) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: Copy paste mistake? 😆

Comment (devlead) [inline src/Cake.Core/Configuration/CakeConfiguration.cs]: Corrected 😊

**Comment by devlead** [comment]
@patriksvensson @cake-build/cake-team feedback should be addressed and rebased against latest.

**Pull request merged**